### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by LaurentMazare** [LaurentMazare/ocaml-wasmtime#1]
@phated and the Grain team got this to work using this [dune discover file](https://github.com/grain-lang/wasmtime.ml/blob/first-pass/src/config/discover.ml).
This could probably be used here too.

**Comment by phated** [comment]
GitHub Actions is a great way to try this on Windows too!

**Comment by LaurentMazare** [comment]
That's a very good point thanks, I was waiting on the libwasmtime opam package to be available in order to set these up as this would solve the issue of getting the c library installed (but obviously this will require a libwasmtime...windows package to get this convenient for windows CI).

**Comment by phated** [comment]
Wasmtime actually hosts all their build packages on releases https://github.com/bytecodealliance/wasmtime/releases/tag/v0.21.0 - this includes Windows & MinGW artifacts!

**Comment by LaurentMazare** [comment]
Yes that's actually what the libwasmtime opam packages are using, they download the appropriate file and uncompress it at a location such that the discover script can find the headers/lib.
The current opam PR only contains the linux and macos version as these extract the archive via some shell calls (using tar) and I'm a bit unsure whether this would work on windows (using unzip in this case).
